cmd/zt_sync_upload_windows_test.go

```diff
@@ -0,0 +1,125 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	chk "gopkg.in/check.v1"
+)
+
+func (s *cmdIntegrationSuite) TestSyncUploadWithExcludeAttrFlag(c *chk.C) {
+	bsu := getBSU()
+
+	srcDirName := scenarioHelper{}.generateLocalDirectory(c)
+	fileList := scenarioHelper{}.generateCommonRemoteScenarioForLocal(c, srcDirName, "")
+
+	// add special files with attributes that we wish to exclude
+	filesToExclude := []string{"file1.pdf", "file2.txt", "file3"}
+	scenarioHelper{}.generateLocalFilesFromList(c, srcDirName, filesToExclude)
+	attrList := []string{"H", "I", "C"}
+	excludeAttrsStr := "H;I;S"
+	scenarioHelper{}.setAttributesForLocalFiles(c, srcDirName, filesToExclude, attrList)
+
+	// set up the destination as an empty container
+	containerURL, containerName := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+
+	// set up interceptor
+	mockedRPC := interceptor{}
+	Rpc = mockedRPC.intercept
+	mockedRPC.init()
+
+	rawContainerURLWithSAS := scenarioHelper{}.getRawContainerURLWithSAS(c, containerName)
+	raw := getDefaultSyncRawInput(srcDirName, rawContainerURLWithSAS.String())
+	raw.excludeFileAttributes = excludeAttrsStr
+
+	runSyncAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.IsNil)
+		validateUploadTransfersAreScheduled(c, "", "", fileList, mockedRPC)
+	})
+}
+
+func (s *cmdIntegrationSuite) TestSyncUploadWithIncludeAttrFlag(c *chk.C) {
+	bsu := getBSU()
+
+	srcDirName := scenarioHelper{}.generateLocalDirectory(c)
+	scenarioHelper{}.generateCommonRemoteScenarioForLocal(c, srcDirName, "")
+
+	// add special files with attributes that we wish to include
+	filesToInclude := []string{"file1.txt", "file2.pdf", "file3.pdf"}
+	scenarioHelper{}.generateLocalFilesFromList(c, srcDirName, filesToInclude)
+	attrList := []string{"H", "I", "C"}
+	includeAttrsStr := "H;I;S"
+	scenarioHelper{}.setAttributesForLocalFiles(c, srcDirName, filesToInclude, attrList)
+
+	// set up the destination as an empty container
+	containerURL, containerName := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+
+	// set up interceptor
+	mockedRPC := interceptor{}
+	Rpc = mockedRPC.intercept
+	mockedRPC.init()
+
+	rawContainerURLWithSAS := scenarioHelper{}.getRawContainerURLWithSAS(c, containerName)
+	raw := getDefaultSyncRawInput(srcDirName, rawContainerURLWithSAS.String())
+	raw.includeFileAttributes = includeAttrsStr
+
+	runSyncAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.IsNil)
+		validateUploadTransfersAreScheduled(c, "", "", filesToInclude, mockedRPC)
+	})
+}
+
+// Asserting that name filter and attribute filter are ANDed
+// Create one file that matches only the name filter
+// Create one file that matches only the attribute filter
+// Create one file that matches both
+// Only the last file should be transferred
+func (s *cmdIntegrationSuite) TestSyncUploadWithIncludeAndIncludeAttrFlags(c *chk.C) {
```

Comment:
You have an (intended) AND case for includes, could you do the same for excludes?

Reply:
Done. By ANDing exclude filter and exclude attribute filter, **none** of the 3 special files (one matches name filter, one matches attribute filter and one matches both) should be transferred.